### PR TITLE
Add "Be Resilient" section

### PR DESCRIPTION
**Pull request by brandur**
Adds a section on making services resilient.

A partial fix for #6.

@tt Can you review this one? Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,6 +122,40 @@ Services should instrument the time that it takes them to run and the time they
...
+
+Harden services by considering when their dependencies might become unavailable and using circuit breakers to protect themselves against degraded external conditions. Use rate limiting and control rods to build layers of defense.
+
+---
+
+Identify single points of failure and compensate for their unavailability where possible. For example, a service might not be able to make much progress anyway if it's database becomes totally unavailable, but should be able to stay online if its ephemeral Redis gets knocked offline.
+
+### Circuit Breakers
+
+The _circuit breaker_ pattern identifies a certain number of failures from an external service within a configured period of time. During such an event the circuit breaker "opens" so that subsequent requests to that service fail automatically and cheaply without having to wait on an expensive timeout. An external service in this sense might be another service with the platform, or an attached resource like Redis.
```
Comment (tt) [inline README.md:135]: This is pretty implementation specific but I feel often circuit breakers do the wrong thing by opening the circuit even for failures that are "safe". There is a huge difference between a service timing out and one that timely produces error responses.

Comment (brandur) [reply README.md:135]: Yeah, I think it's up to the engineers in charge to properly differentiate a condition that's likely to be sustained versus one that's likely to not be. Another paragraph in here about that couldn't hurt though.

Comment (tt) [reply README.md:135]: I can add something. My contributions are long overdue.

Comment (brandur) [reply README.md:135]: Cool. Go for it, and lemme know.

This message just spurred me to go re-read the content we have in here and I think there's quite a lot of good material already (I even stole a section and turned it into a blog post for Stripe) — I wouldn't mind sinking some more time in and pushing it across the finish line.

**Pull request merged**